Add CLDR pluralization options

msfstef opened the pull request:
I have been using Phrase to manage my app translations and they seem to use the CLDR Plural Rules, which have "zero", "one" and "other" as the plural form keys.

It would be nice to include these as well as it doesn't affect the rest of the library and only broadens the support with little overhead.

msfstef committed:
Additional pluralization format

CLDR pluralization rule: http://cldr.unicode.org/index/cldr-spec/plural-rules

msfstef committed:
Update README to include CLDR pluralization

billneff79 commented:
Thanks for the contribution @msfstef .  Seems reasonable.  I have a couple of comments and then we'll get this merged.

billneff79 commented on src/lib/translate.js:
```diff
@@ -23,13 +23,14 @@ export default function translate(id, scope, dictionary, fields, plural, fallbac
 		if (value.splice) {
 			value = value[plural] || value[0];
 		}
 		else if (plural===0 && defined(value.none)) {
```
This conditional should take into account whether `value.none` or `value.zero` are defined, similar to the next conditional

msfstef committed:
Check if "zero" pluralisation form is defined

msfstef committed:
Some test cases for pluralisation

Checking the main pluralisation conventions work

msfstef committed:
Fix pluralisation tests

msfstef commented:
Fixed conditional and added some test cases.

I currently use preact-X in my project, but figured I'd make this request to master first. Should I subsequently make a PR to preact-X, or do you update both in sync?

Thank you

billneff79 commented:
This looks great.  It should be a pretty straight merge to the preact-x branch. I'll take care of that and get a release out for both branches.  Thanks for the commit!
